src/cpp/src/llm_pipeline.cpp

```diff
@@ -89,24 +90,36 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
         const ov::genai::Tokenizer& tokenizer,
         const std::string& device,
         const ov::AnyMap& plugin_config
-    ): 
+    ):
         LLMPipelineImplBase(tokenizer, utils::from_config_json_if_exists(model_path))
     {
         ov::Core core;
-        core.set_property(device, plugin_config);
-        m_model_runner = core.compile_model(model_path / "openvino_model.xml", device).create_infer_request();
+        auto adapters_iter = plugin_config.find(ov::genai::adapters.name());
+        if (adapters_iter != plugin_config.end()) {
+            m_generation_config.adapters = adapters_iter->second.as<AdapterConfig>();
+            auto filtered_plugin_config = plugin_config;
+            filtered_plugin_config.erase(ov::genai::adapters.name());
+            core.set_property(device, filtered_plugin_config);
+            auto model = core.read_model(model_path / "openvino_model.xml");
+            m_adapter_controller = AdapterController(model, m_generation_config.adapters, "base_model.model.model.", device);   // TODO: Make the prefix name configurable
+            m_model_runner = core.compile_model(model, device).create_infer_request();
+            m_adapter_controller->apply(m_model_runner, m_generation_config.adapters);
+        } else {
+            core.set_property(device, plugin_config);
```

Comment:
it's better to pass `plugin_config` / `filtered_plugin_config` to `compile_model` call

Reply:
Did that and part of the tests failed with `NotFound: Unsupported property ENABLE_MMAP by CPU plugin`. Looks like not all the properties can be passed to `compile_model`. In case when `read_model` and `compile_model` are called separately it is likely expected behavior I think. But in case when both is called in a single `compile_model`, it should be fixed at the OV core level. @praasz ?